hydromt/raster.py

```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
     def _check_dimensions(self) -> None:
         """Validate the number and order of dimensions."""
         dims = (self.y_dim, self.x_dim)
-        da = self._obj[self.vars[0]] if isinstance(self._obj, xr.Dataset) else self._obj
-        extra_dims = [dim for dim in da.dims if dim not in dims]
+
+        def _check_dims(da: xr.DataArray, dims=dims) -> bool:
+            extra_dims = [dim for dim in da.dims if dim not in dims]
+            if len(extra_dims) == 1:
+                dims = tuple(extra_dims) + dims
+            elif len(extra_dims) > 1:
+                raise ValueError("Only 2D and 3D data arrays supported.")
+            return da.dims == dims, extra_dims
```

Comment:
Please don't use closures if they are not necessary, they can cause very unexpected behavour. Could you also give this function  a more descriptive name that includes what it actually checks?

Reply:
Also your type hint is incorrect, it returns a bool and I think a list of dimentions?

Reply:
good suggestion